.github/workflows/ci.yaml

```diff
@@ -0,0 +1,61 @@
+# This file is part of Edgehog.
+#
+# Copyright 2024 - 2025 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: ci
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - master
+      - release-*
+permissions:
+  contents: read
+# Spend CI time only on latest ref
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: bash
+    working-directory: rust
+jobs:
+  dco:
+    uses: ./.github/workflows/dco.yaml
+    permissions:
+      actions: read
+      pull-requests: read
+    with:
+      pr: ${{ github.event.pull_request.number }}
+  reuse:
+    uses: ./.github/workflows/reuse-lint.yaml
+  protobuf:
+    uses: ./.github/workflows/protobuf.yaml
+  warmup-caches:
+    uses: ./.github/workflows/warmup-caches.yaml
+    needs: [reuse, dco]
+  codegen:
+    uses: ./.github/workflows/codegen.yaml
+    needs: [warmup-caches]
+  check:
+    uses: ./.github/workflows/check.yaml
+    needs: [warmup-caches]
+  test:
+    uses: ./.github/workflows/test.yaml
+    secrets: inherit
+    needs: [warmup-caches]
```

Comment:
I would probably prepare this CI to the addition of C++.
Using prefixes for the Rust specific workflows